sycl/doc/extensions/proposed/sycl_ext_oneapi_raw_kernel_arg.asciidoc

```diff
@@ -0,0 +1,147 @@
+= sycl_ext_oneapi_raw_kernel_arg
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+:endnote: &#8212;{nbsp}end{nbsp}note
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2023-2023 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 8 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+
+== Overview
+
+When launching kernels that are represented as `sycl::kernel` objects,
+developers must pass arguments via the `set_arg` or `set_args` functions. These
+functions accept trivially copyable types.
```

Comment:
This last sentence is not true.  The existing `set_arg` and `set_args` functions can also take kernel parameters that are not trivially copyable.  For example, you can pass an `accessor` to these functions.

I agree, though, that the new `raw_kernel_arg` wrapper can only be used to pass kernel arguments that are trivially copyable.

Reply:
You're right, I don't know what I was thinking here.

Do you think the below is accurate, or should we just drop the sentence and say nothing?

```suggestion
functions only accept types adhering to the rules defined in
Section 4.12.4 of the SYCL specification.
```

What I'm trying to get at here is that there's no way to pass in an arbitrary number of bytes.

Reply:
That sentence is accurate, but I don't think it's necessary to say that in this specification.  Therefore, I think we could just drop this sentence.  See below, though, for something I think we should add.